From f6e45a443f765731efcd791c52df9d5544da5985 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Sun, 17 Apr 2022 19:50:32 +0000
Subject: [PATCH] WebOSServiceBridgeInjection: fix JSON input test

Passing a JSON array as a parameter to a PalmServiceBridge call would
crash webruntime.

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
---
 .../renderer/webosservicebridge/webosservicebridge_injection.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/neva/injection/renderer/webosservicebridge/webosservicebridge_injection.cc b/src/neva/injection/renderer/webosservicebridge/webosservicebridge_injection.cc
index c3284a1664..55a7ea9900 100644
--- a/src/neva/injection/renderer/webosservicebridge/webosservicebridge_injection.cc
+++ b/src/neva/injection/renderer/webosservicebridge/webosservicebridge_injection.cc
@@ -52,7 +52,7 @@ const char kMethodInvocationAsConstructorOnly[] =
 
 bool IsSubscription(const std::string& param) {
   base::Optional<base::Value> json = base::JSONReader::Read(param);
-  if (!json && !json->is_dict())
+  if (!json || !json->is_dict())
     return false;
 
   const base::Value* subscribe_value =
-- 
2.17.1

